core/src/main/java/tc/oc/pgm/goals/ShowOptions.java

```diff
@@ -0,0 +1,45 @@
+package tc.oc.pgm.goals;
+
+import java.util.HashSet;
+import java.util.Set;
+import org.jdom2.Element;
+import tc.oc.pgm.util.xml.InvalidXMLException;
+import tc.oc.pgm.util.xml.XMLUtils;
+
+public class ShowOptions {
```

Comment:
I think this should just be a top-level `enum`.

```java
public enum ObjectiveOption {
   SHOW_MESSAGES("show-messages")
   // ...
}
```

Parsing code should exist exclusively in `XMLUtils`.

Reply:
> Parsing code should exist exclusively in `XMLUtils`.

@Electroid How should the implementation be handled for moving parsing to  `XMLUtils`. Should `ObjectiveOption` be moved as well so we can reference it directly and have the method be something like:
```java
public static Set<ObjectiveOption> parseObjectiveOptions(Element el, boolean def) {
// ...
}
```
Or were you thinking of a more generic parsing method? I feel like since this is a pretty specific use case it's better off residing inside of `ObjectiveOptions`, similar to how we already have a `KitParser` or `ControlPointParser`. If you want I can extract the static parse method to a separate `ObjectiveOptionParser` class, but seems unnecessary. 

Any suggestions would be appreciated 👍

Reply:
> I think this should just be a top-level `enum`.
Yhe thing is "ShowOptions" is not an enum, it's a set of show flags. and it makes sense for them to be wrapped rather than having to pass around an `ImmutableSet<ObjectiveOption>`

Also, i have one major issue with naming it "objective option" and that is that it's an over-generalization, this is specifically for "show-*" toggles, not for any random boolean you want to put on an objective, otherwise we'd shove half of the objective xml into here